core/src/main/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartition.java

```diff
@@ -139,6 +139,7 @@ public void clear() throws ObjectStoreException {
     synchronized (realKeyToUUIDIndex) {
       try {
         cleanDirectory(this.partitionDirectory);
+        open();
```

Comment:
I think you only need the `createOrRetrievePartitionDescriptorFile();` part of the open

Reply:
also, does it make sense to delete the descriptor in the first place? or instead we need to whitelist the descriptor file?

Reply:
Check if mule 3 doesn't have the same bug. My guess is it does